lcls_tools/common/devices/device.py

```diff
@@ -141,3 +210,37 @@ def remove_callback_from_pv(self, pv: str, function: Callable):
         index = self._get_callback_index(pv_obj, function)
         if index:
             pv_obj.remove_callback(index)
+
+
+class DeviceCollection(BaseModel):
+    devices: Dict[str, SerializeAsAny[Device]] = None
+
+    @field_validator("devices", mode="before")
```

Comment:
@MattKing06 should this be changed or stay as is?

Reply:
I've made this change and there is no difference in functionality, so I'm happy to change it. Makes sense since I'm only changing the dictionary not making any objects or anything.

Reply:
I think this needs to stay as is, I've made the change and run unit tests which failed with this change.

It comes down to the fact that we are storing a dictionary of Devices here, if we were not constructing the device objects here, and had a `Dict[str, Dict[str, Any]]` then we could have an `after` validator but the object construction forces us to manipulate the incoming dictionary before creating `Device` objects to store in the dict.